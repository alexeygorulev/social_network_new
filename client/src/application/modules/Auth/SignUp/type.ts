import { Data } from '../types';

export type SignUpProps = Readonly<{
  data: Data;
  handleChange: () => void;
  handleSubscribe: () => void;
}>;

export type InputItem = Readonly<{
  id: string;
  label: string;
  value: string;
}>;
