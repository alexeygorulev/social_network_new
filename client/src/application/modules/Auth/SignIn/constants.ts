export const labels = {
  sigIn: 'Войдите в аккаунт',
  forgetPassword: 'Забыли пароль?',
  sigUp: 'Зарегистрироваться',
  login: 'Логин',
  password: 'Пароль',
  enterButton: 'Войти',
};

export const fieldsSignIn = {
  username: 'username',
  password: 'password',
};
