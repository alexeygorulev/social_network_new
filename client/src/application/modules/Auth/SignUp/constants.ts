export const fieldsSignUp = {
  firstName: 'firstName',
  middleName: 'middleName',
  lastName: 'lastName',
  login: 'login',
  email: 'email',
  password: 'password',
  duplicatePassword: 'duplicatePassword',
};

export const labels = {
  [fieldsSignUp.firstName]: 'Имя',
  [fieldsSignUp.middleName]: 'Фамилия',
  [fieldsSignUp.lastName]: 'Отчество',
  [fieldsSignUp.login]: 'Ваш Логин',
  [fieldsSignUp.email]: 'Ваш Email',
  [fieldsSignUp.password]: 'Ваш пароль',
  [fieldsSignUp.duplicatePassword]: 'Повторите пароль',
};

export const titles = {
  main: 'Заполните данные',
  button: 'Отправить',
};
