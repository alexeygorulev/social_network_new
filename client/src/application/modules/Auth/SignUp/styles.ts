import styled from 'styled-components';

export const StyledBackToAuthIcon = styled.div`
  position: absolute;
  width: 20px;
  height: 20px;
  cursor: pointer;
  z-index: 1002;
`;
