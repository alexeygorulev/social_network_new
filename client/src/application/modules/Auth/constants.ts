export const labels = {
  logIn: 'Войдите в аккаунт',
};

export const fields = {
  username: 'username',
  password: 'password',
};
