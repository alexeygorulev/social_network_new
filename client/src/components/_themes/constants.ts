export const mediaBreakpoints = {
  largePhone: 480,
  tablet: 768,
  smallDesktop: 992,
  desktop: 1200,
  largeDesktop: 1440,
};

export const mediaSizes = {
  s: 's', // 320 - 768      / PHONE
  m: 'm', // 768 - 1200     / TABLET
  l: 'l', // 1200 - âˆž       / DESKTOP
};

export const media = {
  largePhone: `min-width: ${mediaBreakpoints.largePhone}px`,
  tablet: `min-width: ${mediaBreakpoints.tablet}px`,
  smallDesktop: `min-width: ${mediaBreakpoints.smallDesktop}px`,
  desktop: `min-width: ${mediaBreakpoints.desktop}px`,
  largeDesktop: `min-width: ${mediaBreakpoints.largeDesktop}px`,
};
