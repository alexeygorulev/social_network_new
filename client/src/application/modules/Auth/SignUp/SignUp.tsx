import Block from 'components/atoms/Block';
import InputText from 'components/inputs/InputText';
import { InputItem, SignUpProps } from './type';
import { labels, titles } from './constants';
import { StyledAuthFormContainer } from '../style';
import Button from 'components/atoms/Button';
import IconArrowLeft from 'components/icons/IconArrowLeft';
import { StyledBackToAuthIcon } from './styles';
import { fieldsSignUp } from '../constants';

const SignUp = (props: SignUpProps) => {
  const { handleChange, data, handleSubscribe, changeStep } = props;
  const inputArray = Object.keys(fieldsSignUp).reduce(
    (result, item) => [...result, { id: item, label: labels[item], value: data.valuesSignUp[item] }],
    [],
  );

  return (
    <StyledAuthFormContainer registration>
      <StyledBackToAuthIcon data-testid="auth" onClick={() => changeStep()}>
        <IconArrowLeft />
      </StyledBackToAuthIcon>
      <Block textAlign="center">{titles.main}</Block>
      {inputArray.map((input: InputItem) => (
        <Block key={input.id} margin="s">
          <InputText id={input.id} label={input.label} value={input.value} onChange={handleChange} />
        </Block>
      ))}
      <Block margin="l" textAlign="center">
        <Button id="submit" onClick={handleSubscribe}>
          {titles.button}
        </Button>
      </Block>
    </StyledAuthFormContainer>
  );
};

export default SignUp;
