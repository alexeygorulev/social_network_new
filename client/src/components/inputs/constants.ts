import { mediaSizes } from 'components/_themes/constants';

export const iconProps = {
  right: {
    [mediaSizes.s]: { s: 8, m: 16, l: 16 },
    [mediaSizes.m]: { s: 8, m: 16, l: 16 },
    [mediaSizes.l]: { s: 8, m: 16, l: 16 },
  },
};
