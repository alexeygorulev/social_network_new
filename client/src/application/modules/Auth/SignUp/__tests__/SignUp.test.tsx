import SignUp from '../SignUp';
import { fieldsSignUp } from '../constants';
import { renderWithTheme } from 'utilsTests';
import { fireEvent, screen } from '@testing-library/react';

describe('SignUp.tsx: ', () => {
  const mountMock = jest.fn();
  const unmountMock = jest.fn();
  const onCheckLoginUserMock = jest.fn();
  const onSubscribeUserMock = jest.fn();
  const changeStepMock = jest.fn();
  const handleChangeMock = jest.fn();
  const signUpStepMock = jest.fn();
  const signInStepMock = jest.fn();

  const valuesSignUp = Object.values(fieldsSignUp).reduce((result, item) => ({ ...result, [item]: '' }), {});

  const setUp = () => ({
    mount: mountMock,
    unmount: unmountMock,
    data: { valuesSignIn: { username: '', password: '' }, valuesSignUp },
    handleChange: handleChangeMock,
    isLoading: false,
    onCheckLoginUser: onCheckLoginUserMock,
    onSubscribeUser: onSubscribeUserMock,
    changeStep: changeStepMock,
    signUpStep: signUpStepMock,
    signInStep: signInStepMock,
  });

  afterEach(() => {
    handleChangeMock.mockClear();
    onCheckLoginUserMock.mockClear();
    changeStepMock.mockClear();
  });

  it('should match the snapshot', () => {
    const { data, signUpStep, handleChange, onSubscribeUser } = setUp();
    const { asFragment } = renderWithTheme(
      <SignUp data={data} changeStep={signUpStep} handleChange={handleChange} handleSubscribe={onSubscribeUser} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('testing handleChange', () => {
    const { data, signUpStep, handleChange, onSubscribeUser } = setUp();

    renderWithTheme(
      <SignUp data={data} changeStep={signUpStep} handleChange={handleChange} handleSubscribe={onSubscribeUser} />,
    );

    const emailInput = screen.getByTestId(fieldsSignUp.email) as HTMLInputElement;
    fireEvent.change(emailInput, { target: { value: 'asd' } });
    expect(handleChangeMock).toHaveBeenCalledWith({
      id: fieldsSignUp.email,
      value: 'asd',
      formattedValue: 'asd',
    });
  });

  it('testing button click', () => {
    const { data, signUpStep, handleChange, onSubscribeUser } = setUp();

    renderWithTheme(
      <SignUp data={data} changeStep={signUpStep} handleChange={handleChange} handleSubscribe={onSubscribeUser} />,
    );

    const buttonClick = screen.getByTestId('submit') as HTMLInputElement;
    fireEvent.click(buttonClick);
    expect(onSubscribeUser).toHaveBeenCalledTimes(1);
  });

  it('should call changeStep when Span with cursor pointer is clicked', () => {
    const { data, signInStep, handleChange, onSubscribeUser } = setUp();

    renderWithTheme(
      <SignUp data={data} changeStep={signInStep} handleChange={handleChange} handleSubscribe={onSubscribeUser} />,
    );

    const spanClick = screen.getByTestId('auth') as HTMLInputElement;
    fireEvent.click(spanClick);
    expect(signInStep).toHaveBeenCalledTimes(1);
  });
});
