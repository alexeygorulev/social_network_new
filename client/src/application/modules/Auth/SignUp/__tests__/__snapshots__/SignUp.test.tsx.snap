// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp.tsx:  should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-kDnyCx gULQLa"
  >
    <div
      class="sc-idyqAC ieWywR"
      data-testid="auth"
    >
      <div
        class="sc-ksJisA hdUcFu"
        data-component="Icon"
      >
        <svg
          class="sc-hBpgZr gdtCmN"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <svg
            id="Layer_1"
            version="1.1"
            viewBox="0 0 492 492"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <g>
              <g>
                <path
                  d="M464.344,207.418l0.768,0.168H135.888l103.496-103.724c5.068-5.064,7.848-11.924,7.848-19.124 c0-7.2-2.78-14.012-7.848-19.088L223.28,49.538c-5.064-5.064-11.812-7.864-19.008-7.864c-7.2,0-13.952,2.78-19.016,7.844 L7.844,226.914C2.76,231.998-0.02,238.77,0,245.974c-0.02,7.244,2.76,14.02,7.844,19.096l177.412,177.412 c5.064,5.06,11.812,7.844,19.016,7.844c7.196,0,13.944-2.788,19.008-7.844l16.104-16.112c5.068-5.056,7.848-11.808,7.848-19.008 c0-7.196-2.78-13.592-7.848-18.652L134.72,284.406h329.992c14.828,0,27.288-12.78,27.288-27.6v-22.788 C492,219.198,479.172,207.418,464.344,207.418z"
                />
              </g>
            </g>
          </svg>
        </svg>
      </div>
    </div>
    <div
      class="sc-grXZZQ igFxnq"
      data-component="Block"
    >
      Заполните данные
    </div>
    <div
      class="sc-grXZZQ bHNNuL"
      data-component="Block"
    >
      <div
        class="sc-dkjKgF jyXmYe"
      >
        <input
          class="sc-eDnVMP sc-gKHVLF hWqSJa"
          data-testid="firstName"
          id="firstName"
          inputmode="text"
          value=""
        />
        <label
          class="sc-hLclGa duLwaM"
        >
          Имя
        </label>
      </div>
    </div>
    <div
      class="sc-grXZZQ bHNNuL"
      data-component="Block"
    >
      <div
        class="sc-dkjKgF jyXmYe"
      >
        <input
          class="sc-eDnVMP sc-gKHVLF hWqSJa"
          data-testid="middleName"
          id="middleName"
          inputmode="text"
          value=""
        />
        <label
          class="sc-hLclGa duLwaM"
        >
          Фамилия
        </label>
      </div>
    </div>
    <div
      class="sc-grXZZQ bHNNuL"
      data-component="Block"
    >
      <div
        class="sc-dkjKgF jyXmYe"
      >
        <input
          class="sc-eDnVMP sc-gKHVLF hWqSJa"
          data-testid="lastName"
          id="lastName"
          inputmode="text"
          value=""
        />
        <label
          class="sc-hLclGa duLwaM"
        >
          Отчество
        </label>
      </div>
    </div>
    <div
      class="sc-grXZZQ bHNNuL"
      data-component="Block"
    >
      <div
        class="sc-dkjKgF jyXmYe"
      >
        <input
          class="sc-eDnVMP sc-gKHVLF hWqSJa"
          data-testid="login"
          id="login"
          inputmode="text"
          value=""
        />
        <label
          class="sc-hLclGa duLwaM"
        >
          Ваш Логин
        </label>
      </div>
    </div>
    <div
      class="sc-grXZZQ bHNNuL"
      data-component="Block"
    >
      <div
        class="sc-dkjKgF jyXmYe"
      >
        <input
          class="sc-eDnVMP sc-gKHVLF hWqSJa"
          data-testid="email"
          id="email"
          inputmode="text"
          value=""
        />
        <label
          class="sc-hLclGa duLwaM"
        >
          Ваш Email
        </label>
      </div>
    </div>
    <div
      class="sc-grXZZQ bHNNuL"
      data-component="Block"
    >
      <div
        class="sc-dkjKgF jyXmYe"
      >
        <input
          class="sc-eDnVMP sc-gKHVLF hWqSJa"
          data-testid="password"
          id="password"
          inputmode="text"
          value=""
        />
        <label
          class="sc-hLclGa duLwaM"
        >
          Ваш пароль
        </label>
      </div>
    </div>
    <div
      class="sc-grXZZQ bHNNuL"
      data-component="Block"
    >
      <div
        class="sc-dkjKgF jyXmYe"
      >
        <input
          class="sc-eDnVMP sc-gKHVLF hWqSJa"
          data-testid="duplicatePassword"
          id="duplicatePassword"
          inputmode="text"
          value=""
        />
        <label
          class="sc-hLclGa duLwaM"
        >
          Повторите пароль
        </label>
      </div>
    </div>
    <div
      class="sc-grXZZQ NQnJH"
      data-component="Block"
    >
      <button
        class="sc-fnOeiS jGftQW"
        data-component="Button"
        data-testid="submit"
        id="submit"
        type="button"
      >
        <div
          class="sc-fEyylQ iaGUbZ"
        >
          Отправить
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
